tests/functional/object/mcg/test_s3_vector.py

```
@@ -1310,3 +1331,592 @@ def test_vector_distance_metrics(self, vector_bucket_factory, mcg_obj):
+        ), f"Old rule AllowListBucket still present after update. Sids: {set(updated_stmts)}"
+        assert (
+            "DenyDeleteVector" not in updated_stmts
+        ), f"Old rule DenyDeleteVector still present after update. Sids: {set(updated_stmts)}"
+        assert (
+            updated_stmts["AllowQueryVectors"].get("Effect") == "Allow"
+        ), f"AllowQueryVectors Effect mismatch: {updated_stmts['AllowQueryVectors']}"
+        assert (
+            updated_stmts["DenyPutVectors"].get("Effect") == "Deny"
+        ), f"DenyPutVectors Effect mismatch: {updated_stmts['DenyPutVectors']}"
+        logger.info(
+            f"✓ Updated policy verified — new sids: {set(updated_stmts)}, old rules removed"
```

Ruff (0.15.18)

[warning] Logging statement uses f-string

(G004)